contract/sep-nft-collection.md

```diff
@@ -0,0 +1,114 @@
+## Summary
+
+This proposal introduces a standard interface for creating and managing NFT collections on Soroban, ensuring compatibility and interoperability within the Stellar ecosystem, and aligning with the previously defined NFT interface and the ERC-1155 standard.
+
+## Motivation
+
+A standardized interface for NFT collections on Soroban will facilitate seamless interaction within the Stellar ecosystem, enhancing user experience and developer interaction by providing a consistent API for NFT collection creation, management, and interaction, while also allowing for efficient batch transfers and management of both fungible and non-fungible tokens within a single contract.
+
+## Abstract
+
+This proposal defines a Rust trait for NFT collections, ensuring that all NFT collection contracts on Soroban adhere to a common API. This API includes functions for creating new NFTs within a collection, transferring NFTs, querying for ownership, and for accessing metadata necessary for displaying NFT information in wallets and other user interfaces.
+
+## Specification
+
+```rust
+pub trait NFTCollectionFactory {
+    // Admin interface – privileged functions.
+    fn initialize(
+        env: Env, 
+        admin: Address, 
+        collection_name: String, 
+        collection_symbol: String
+    );
+
+    fn mint_nft(
+        env: Env, 
+        to: Address, 
+        name: String, 
+        symbol: String, 
+        royalty_recipient: Address, 
+        royalty_percentage: u32, 
```

Comment:
Is this in basis points, e.g. 0-10,000?

Reply:
> Is this in basis points, e.g. 0-10,000?

I guess it should be defined because but it is meant to be a percentage as a full number.

Basically 1 over royalty_p

  Im not sure if i should have even included royalty metadat in these interfaces as in eth it is a whole seperate eip however on eth its up to the marketplace to actually comply with royalties so if there is a way to force compliance the info would need to be accessible in all i think. 

Maybe theres a better way?

Reply:
Perhaps the metadata could include it as on option type? Then have the mint methods take metadata as an argument instead of an argument for each field.